Fix "make uImage" load and entry addresses in 32-bit mode.

Signed-off-by: Chris Smith <chris.smith@st.com>
Index: linux-sh4-2.6.23.16_stm23_0109/arch/sh/boot/Makefile
===================================================================
--- linux-sh4-2.6.23.16_stm23_0109.orig/arch/sh/boot/Makefile	2008-03-26 12:26:34.000000000 +0000
+++ linux-sh4-2.6.23.16_stm23_0109/arch/sh/boot/Makefile	2008-03-28 10:40:29.000000000 +0000
@@ -36,10 +36,17 @@
 $(obj)/compressed/vmlinux: FORCE
 	$(Q)$(MAKE) $(build)=$(obj)/compressed $@
 
+ifeq ($(CONFIG_32BIT),y)
+KERNEL_LOAD	:= $(shell /bin/bash -c 'printf "0x%8x" \
+		     $$[$(CONFIG_PAGE_OFFSET)  + \
+			$(CONFIG_ZERO_PAGE_OFFSET)]')
+else
 KERNEL_LOAD	:= $(shell /bin/bash -c 'printf "0x%8x" \
 		     $$[$(CONFIG_PAGE_OFFSET)  + \
 			$(CONFIG_MEMORY_START) + \
 			$(CONFIG_ZERO_PAGE_OFFSET)]')
+endif
+
 KERNEL_ENTRY	:= $(shell /bin/bash -c 'printf "0x%8x" \
 		     $$[$(KERNEL_LOAD)+0x1000]')
 
